scripts/eqtl_hail_batch/launch_generate_eqtl_spearman.py

```diff
@@ -114,40 +117,39 @@ def file_exists(files):
                 # check all files exist before running
                 files_to_check = [geneloc]
                 files_that_are_missing = filter(
-                    lambda x: not file_exists(x), files_to_check
+                    lambda x: not hl.hadoop_exists(x), files_to_check
                 )
                 for file in files_that_are_missing:
                     logging.error(f'File {file} is missing')
             else:
 
+                job = batch.new_job(f'{cell_type}-chr{chromosome}')
+                copy_common_env(job)
+                job.image(get_config()['workflow']['driver_image'])
+                
+                # check out a git repository at the current commit
+                prepare_git_job(
+                    job=job,
+                    organisation=get_organisation_name_from_current_directory(),
+                    repo_name=get_repo_name_from_current_directory(),
+                    commit=get_git_commit_ref_of_current_repository(),
+                )
+
+                keys = os.path.join(input_path, 'OneK1K_CPG_IDs.tsv')
                 output_prefix = os.path.join(
                     output_dir, f'{cell_type}', f'chr{chromosome}'
                 )
-                # add in sampleid keys
-                keys = os.path.join(input_path, 'OneK1K_CPG_IDs.tsv')
-                # The analysis-runner output path doesn't want the BUCKET specified,
-                # so let's remove it from the output_prefix
-                analysis_runner_output_path = output_prefix[5:].partition('/')[-1]
-                # get access level from bucket, rather than manual input
-                access_level = bucket_name.split('-')[-1]
-                if access_level == 'main':
-                    access_level = 'standard'
-                run_analysis_runner(
-                    description=f'eqtl_spearman_{cell_type}_chr{chromosome}',
-                    dataset='tob-wgs',
-                    access_level=access_level,
-                    output_dir=analysis_runner_output_path,
-                    # commit, sha and cwd can be inferred automatically
-                    script=[
-                        'generate_eqtl_spearman.py',
-                        *('--expression', expression),
-                        *('--covariates', covariates),
-                        *('--geneloc', geneloc),
-                        *('--keys', keys),
-                        *('--output-prefix', output_prefix),
-                    ],
+                job.command(
```

Comment:
This looks fine to me, but @illusional suggested that it might be easier to call this directly.

Is this performing much work? If not, could you just call a Python function from this module inline here to construct a job for a larger batch, instead of the indirection through either another batch or calling `subprocess`?

Certainly not a big issue though, so feel free to ignore / revisit in follow-up PR.

Reply:
If this is working ok I might leave as is, then ask @illusional about it for a follow-up PR